syntax = "proto3";
package darkanddarker;

import "player.proto";

message Listing {

    int32 listingId = 1;
    Item item = 2;
    int32 price = 3;
    int32 timeToExpire =4;
    Player seller =5;
}


message Item { 
    
    uint64 itemUniqueId = 1;
    string itemId = 2;
    repeated ItemProperty primaryPropertyArray = 9;
    repeated ItemProperty secondaryPropertyArray = 10;
    int32 itemCount =3;
    int32 unknown2 =4;
    int32 unknown3 =5;
    int32 unknown4 =11;
}

message ItemProperty {
    string property = 1;
    int32 value = 2;
}

message MarketResponse {
    repeated Listing listingArray = 1;
}


message MarketListingRequest {
    repeated QueryProperty queryField = 1;
}

message QueryProperty {
    int32 vaue = 1;
    repeated string key = 2;
}